src/dune_rpc_impl/server.ml

```diff
@@ -204,13 +205,28 @@ let handler (t : t Fdecl.t) : 'a Dune_rpc_server.Handler.t =
     Handler.implement_notification rpc Procedures.Public.shutdown shutdown
   in
   let () =
+    let svar = Build_system.progress () in
+    let module Poll_comparable = Comparable.Make (Poller) in
+    let module Map = Poll_comparable.Map in
+    let pollers = ref Map.empty in
     Handler.implement_poll rpc Procedures.Poll.progress
       ~on_cancel:(fun _session poller ->
         let p = Long_poll.progress (Fdecl.get t).long_poll in
         Long_poll.Instance.client_cancel p poller)
```

Comment:
We need something else here. We could store create and store ivars in the poller map and then just fill it oursleves when the cancellation request comes. That's probably the simplest.